Python/compile.c

```diff
@@ -6175,20 +6175,44 @@ compiler_pattern_mapping(struct compiler *c, pattern_ty p, pattern_context *pc)
     }
     // Collect all of the keys into a tuple for MATCH_KEYS and
     // COPY_DICT_WITHOUT_KEYS. They can either be dotted names or literals:
+
+    // will hold only Constant_kind keys
+    PyObject *seen = PySet_New(NULL);
+    if (seen == NULL) {
+        return -1;
```

Comment:
All of the visitors in the compiler return `0` on failure and `1` on success:

```suggestion
        return 0;
```